### ISSUE TITLE
Snapshots no longer work from 1.4.1 to 1.4.2

### ISSUE THREAD

**Issue by lukas-gust** [dbt-athena/dbt-athena#227]
# Background
I have a dbt project using dbt-athena-community==1.4.1 and dbt-core==1.4.4 running from a docker container in production no problem, i.e. the snapshots are working as intended.

When switching to 1.4.2 and 1.4.5 respectively I receive the following error for all of my snapshot queries.

# Steps to reproduce
1. Run snapshot models with dbt-athena-community==1.4.1 and dbt-core==1.4.4
2. Upgrade both libraries to 1.4.2 and 1.4.5 respectively. (I believe if you just install 1.4.2 of the athena adapter it will upgrade dbt-core)
3. Run snapshots again.

# Error messages and code
```
Runtime Error in snapshot accounts_snapshot (snapshots/onboard_production/accounts_snapshot.sql)
SYNTAX_ERROR: line 48:35: Column 'snapshotted_data.dbt_unique_key' is ambiguous. You may need to manually clean the data at location 's3://<staing-dir-bucket>/<someguid>' before retrying. Athena will not delete data in your account.
```

Here is one of the models, though they are all the same
```
{% snapshot accounts_snapshot %}

{{
   config(
        target_database='awsdatacatalog',
        target_schema='silver_onboard_production_snapshots',
        unique_key='account_id',
        strategy='timestamp',
        updated_at='updated_at',
        partition_by=['dbt_snapshot_at']
   )
}}

select * from {{ ref("accounts") }}


{% endsnapshot %}
```

Here is the failing query in Athena
```
-- /* {"app": "dbt", "dbt_version": "1.4.5", "profile_name": "analytics_dbt", "target_name": "dev", "node_id": "snapshot.silver.accounts_snapshot"} */
        
  create table "silver_onboard_production_snapshots"."accounts_snapshot__dbt_tmp"
  with (
    table_type='hive',
    is_external=true,
    format='parquet'
  )
  as
    
        with snapshot_query as (
        



select * from "silver_onboard_production"."accounts"


    ),
    snapshotted_data as (
        select *,
                unit_id as dbt_unique_key
        from "silver_onboard_production_snapshots"."accounts_snapshot"
            where dbt_valid_to is null
    ),
    insertions_source_data as (
        select
                *,
                unit_id as dbt_unique_key,
                updated_at as dbt_updated_at,
                updated_at as dbt_valid_from,
                nullif(updated_at, updated_at) as dbt_valid_to,
                to_hex(md5(to_utf8(coalesce(cast(unit_id as varchar ), '')
         || '|' || coalesce(cast(updated_at as varchar ), '')
        ))) as dbt_scd_id
        from snapshot_query
        ),
    updates_source_data as (
        select
                *,
                unit_id as dbt_unique_key,
                updated_at as dbt_updated_at,
                updated_at as dbt_valid_from,
                updated_at as dbt_valid_to
        from snapshot_query
        ),
    insertions as (
        select
                'insert' as dbt_change_type,
                source_data.*
        from insertions_source_data as source_data
            left outer join snapshotted_data on snapshotted_data.dbt_unique_key = source_data.dbt_unique_key
            where snapshotted_data.dbt_unique_key is null
               or (
                    snapshotted_data.dbt_unique_key is not null
                and (
                    (snapshotted_data.dbt_valid_from < source_data.updated_at)
                )
            )
    ),
    updates as (
        select
                'update' as dbt_change_type,
                source_data.*,
                snapshotted_data.dbt_scd_id
        from updates_source_data as source_data
            join snapshotted_data on snapshotted_data.dbt_unique_key = source_data.dbt_unique_key
            where (
                (snapshotted_data.dbt_valid_from < source_data.updated_at)
            )
        )
    select * from insertions
        union all
        select * from updates
```

The working query on 1.4.4
```
-- /* {"app": "dbt", "dbt_version": "1.4.4", "profile_name": "analytics_dbt", "target_name": "dev", "node_id": "snapshot.silver.accounts_snapshot"} */

  create table "silver_onboard_production_snapshots"."accounts_snapshot__dbt_tmp"
  with (
    table_type='hive',
    is_external=true,
    external_location='s3://<data lake bucket>//silver/onboard_production/accounts_snapshot__dbt_tmp',
    format='parquet'
  )
  as
    with snapshot_query as (
        



select * from "silver_onboard_production"."accounts"

)
    , snapshotted_data_base as (
        select *
          , ROW_NUMBER() OVER (
              PARTITION BY dbt_unique_key
              ORDER BY dbt_valid_from DESC
            ) as dbt_snapshot_rn
        from "silver_onboard_production_snapshots"."accounts_snapshot"
    )
    , snapshotted_data as (
        select *
        from snapshotted_data_base
        where dbt_snapshot_rn = 1
          AND dbt_change_type != 'delete'
    )
    , source_data as (
        select *
          , account_id as dbt_unique_key
          , updated_at as dbt_valid_from
          , to_hex(md5(to_utf8(coalesce(cast(account_id as varchar ), '')
         || '|' || coalesce(cast(updated_at as varchar ), '')
        ))) as dbt_scd_id
        from snapshot_query
    )
    , upserts as (
        select source_data.*
          , case
              when snapshotted_data.dbt_unique_key IS NULL THEN 'insert'
              else 'update'
            end as dbt_change_type
          , cast('9999-01-01' AS timestamp) as dbt_valid_to
          , True as is_current_record
        from source_data
        LEFT join snapshotted_data
               on snapshotted_data.dbt_unique_key = source_data.dbt_unique_key
        where snapshotted_data.dbt_unique_key IS NULL
           OR (
                snapshotted_data.dbt_unique_key IS NOT NULL
            AND (
                (snapshotted_data.dbt_valid_from < source_data.updated_at)
            )
        )
    )
    select * from upserts
```

**Comment by nicor88** [comment]
Could be potentially address by https://github.com/dbt-athena/dbt-athena/pull/202

**Comment by Jrmyy** [comment]
👋🏻 Hello @lukas-gust 
We have released the version 1.5.0 of the adapter that changes how snapshot works. Can you give it a shot and let us know if it works for you ?
I will close this PR for now since there is no activity, feel free to reopen if needed.
